R/expr-array.R

```diff
@@ -17,6 +17,9 @@ namespace_expr_arr <- function(x) {
   self
 }
 
+# TODO-REWRITE: add in NEWS that some arguments in $arr$join(), $arr$sort(),
+# $arr$unique() now must be named, https://github.com/eitsupi/neo-r-polars/pull/14
```

Comment:
It is better not to mention this in the NEWS because there are too many to mention.
If you use the positional argument, it will not malfunction because it will result in an error.

Reply:
Nevertheless that will convert running code to error so it should be mentioned somewhere. It could be a breaking change or a bug fix, but it should appear in the changelog IMO.

Reply:
In that case would you be ok with an additional page listing all those changes so that we can focus on the most important stuff in the changelog, like https://docs.pola.rs/releases/upgrade/1 ?

I get your point, but even for us it's useful to document when a particular function changed (besides `git blame`).

Reply:
I don't think all the things like argument name changes are listed there.

Wouldn't it be better to modify the current r-polars argument names to cause breaking changes first?

Reply:
I think we can find out later what actually changed. Tracking everything against r-polars at this stage is a waste of time, and I think we should pay more attention to whether it is consistent with py-polars.

Reply:
In other words, I did not look at r-polars when writing methods of each class, only py-polars, so no attention was paid to what is the same and what is different from r-polars.

Reply:
> I think we can find out later what actually changed. 

That's gonna be very hard to find in tens of thousands of lines changed if we don't store this information beforehand. Even if not all is stored I don't think it is a bad thing to keep this info in this kind of comment. Since it's prefixed "TODO-REWRITE" it shouldn't stay hidden in the codebase anyway.

Reply:
> That's gonna be very hard to find in tens of thousands of lines changed if we don't store this information beforehand. Even if not all is stored I don't think it is a bad thing to keep this info in this kind of comment. Since it's prefixed "TODO-REWRITE" it shouldn't stay hidden in the codebase anyway.

As for the change in argument position you pointed out, since it can be extracted with the `formals()` function, why not just write a script to do the comparison for all functions?

Reply:
Yes that's a good idea